File: C++/Basic Hashing/Problem1.cpp

// Count frequency of each element in the array

#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int main() 
{
   int arr[]={10,5,10,15,10,5};
   
   int length=sizeof(arr)/sizeof(arr[0]);
   
   unordered_map<int,int>mapp;
   for(int i=0;i<length;i++){
     mapp[arr[i]]++;
   }
   
   for(auto it : mapp){
     std::cout << it.first<<" "<<it.second << std::endl;
   }
    return 0;
}

File: Javascript/Basic Hashing/Problem1.js


//Count frequency of each element in the array

let arr=[10,5,10,15,10,5];


let map=new Map();

for(let i=0;i<arr.length;i++){
 // console.log(arr[i])
  if(map.has(arr[i])){
    let getval=map.get(arr[i]);
    map.set(arr[i],getval+1);
  }
  else{
    map.set(arr[i],1);
  }
}

for(let [key,value] of map){
  console.log(key,value)
}